### PR TITLE
refactor codes and fix bugs

### PR DESCRIPTION
**Pull request by minggo**
Re: 

### Changelog

* Refactor codes to make it more clear.
* Fix the bug that, not matter what the log level set, log, warn and error can work all the time. After this PR, if the log level is `WARN`, then `log` will not output any information.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/platform/debug.ts [hunk 1]
```diff
@@ -126,104 +126,107 @@ export function debug (...data: any[]) {
...
-            logList.scrollTop = logList.scrollHeight;
+    // Don't change the case order.
+    switch (mode) {
+    case DebugMode.INFO_FOR_WEB_PAGE: {
+        ccLog = (message?: any, ...optionalParams: any[]) => {
+            logToWebPage(formatString(message, ...optionalParams));
         };
-
+    }
+    // eslint-disable-next-line no-fallthrough
```
Comment (shrinktofit) [inline cocos/core/platform/debug.ts]:
```suggestion
    // fallthrough
```

File: cocos/core/platform/debug.ts [hunk 2]
```diff
@@ -126,104 +126,107 @@ export function debug (...data: any[]) {
...
+    case DebugMode.WARN_FOR_WEB_PAGE: {
+        ccWarn = (message?: any, ...optionalParams: any[]) => {
+            logToWebPage(`WARN :  ${formatString(message, ...optionalParams)}`);
+        };
+    }
+    // eslint-disable-next-line no-fallthrough
```
Comment (shrinktofit) [inline cocos/core/platform/debug.ts]:
```suggestion
    // fallthrough
```

File: cocos/core/platform/debug.ts [hunk 3]
```diff
@@ -126,104 +126,107 @@ export function debug (...data: any[]) {
...
+        console.error = console.log;
+    }
+    if (!console.warn) {
+        console.warn = console.log;
+    }
+
+    // Don't change the case order.
+    switch (mode) {
+    case DebugMode.VERBOSE: // Do nothing.
+    // eslint-disable-next-line no-fallthrough
```
Comment (shrinktofit) [inline cocos/core/platform/debug.ts]:
```suggestion
    // fallthrough
```

File: cocos/core/platform/debug.ts [hunk 4]
```diff
@@ -233,25 +236,49 @@ export function _resetDebugSetting (mode: DebugMode) {
...
         }
     }
-
-    if (EDITOR) {
-        ccLog = console.log.bind(console);
-    } else if (mode <= DebugMode.INFO) {
-        if (JSB) {
-            ccLog = console.log;
-        } else if (console.log.bind) {
+    // eslint-disable-next-line no-fallthrough
```
Comment (shrinktofit) [inline cocos/core/platform/debug.ts]:
```suggestion
    // fallthrough
```

File: cocos/core/platform/debug.ts (hunk 1 above)
Comment (minggo) [reply cocos/core/platform/debug.ts]: Good to know. Fixed it.

**Review by SantyWang — APPROVED** [review]

**Pull request merged**